### PR TITLE
Skip k8s processor for events coming from Agent

### PR DESCRIPTION
**Pull request by ChrsMark**
## What does this PR do?
This PR adds a condition to the `add_kubernetes_metadata` processor so as to be skipped in case the event is produced by an Agent setup.

## Why is it important?
So as to avoid having the `add_kubernetes_metadata` processor being enabled by default when Filebeat/Metricbeat is deployed via Elastic Agent.

Closes https://github.com/elastic/beats/issues/29767

### CODE REVIEW

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -83,6 +83,7 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 
 *Affecting all Beats*
 
+- Skip k8s processor for events coming from Agent. {pull}29809[29809]
```
Comment (ruflin) [inline CHANGELOG.next.asciidoc:86]:
```suggestion
- Skip k8s processor for events coming produced by processes run by Elastic Agent. {pull}29809[29809]
```

**Review by ruflin — COMMENTED** [review]
@blakerouse @ph I wonder if there is a nicer way to add this condition for example as part of the transpiler? 

It feels strange that we have to make a change in the default beats configs instead of making it in Elastic Agent.

**Comment by ChrsMark** [comment]
Another option would be to use another default configuration for Beat's config like we do for docker with `filebeat.docker.yaml` https://github.com/elastic/beats/blob/3c864d26bccdb580a2a7985e9fa902f968cda431/filebeat/filebeat.docker.yml. We can have different configs to be used by Beats when spawn by Elastic Agent.

**Comment by ruflin** [comment]
I like this idea a lot. This would potentially also solve some other issues where we use the transpiler to set some defaults instead of just modifying a config file. It would also decouple Beats run under Elastic Agent more cleary. It means we would have an `filebeat.elastic-agent.yml` file?

@cmacknz For awareness.

**Comment by ChrsMark** [comment]
> It means we would have an `filebeat.elastic-agent.yml` file?
> 

Yes. Not sure how easy it would be technically to happen though.

**Comment by blakerouse** [comment]
Simply adding `-c filebeat.elastic-agent.yml` here would do it:

https://github.com/elastic/beats/blob/56423a06169d0bf87769867dd1929518e248e65a/x-pack/elastic-agent/spec/filebeat.yml#L3-L9

**Comment by ph** [comment]
Yes, what is proposed by @blakerouse seems to be the easiest way to have a completely different set of default options.

**Comment by ChrsMark** [comment]
So what could be the next step here? I guess I can close this PR if we agree on going with the `filebeat.elastic-agent.yml` option (I'm +1 for this) and keep https://github.com/elastic/beats/issues/29767 open until we introduce the decoupling? Does this fall to Agent's team ownership?

**Comment by ruflin** [comment]
I think the control plane team should take this over @jlind23 . Seems we all agree the config file would be a better approach.

@blakerouse If we have the separate config, it seems we can get rid of all the other `-E` options and just put it in the config file directly?

**Comment by blakerouse** [comment]
@ruflin Yes we could move them into the configuration file.

**Comment by ph** [comment]
@ruflin @blakerouse I wonder, if this could open a security risk if a user is able to modify the file?

**Comment by blakerouse** [comment]
@ph It is possible that it could allow them to add something, but we already have that today with the default configuration anyway.

**Comment by ruflin** [comment]
I'm closing this PR as I think we all agree we take a different path forward. @ChrsMark Thanks for kicking this off and leading us to the right path.

@ph Could you take the lead in opening up a follow up PR or issue so we can continue the conversation there?

**Pull request closed without merging**